src/SA/get-netsession2/entry.c

```diff
@@ -0,0 +1,305 @@
+                            }
+
+                            if (i != 4)
+                            {
+                                internal_printf("PTR: Failed; Invalid IP address\n");
+                            }
+                            else
+                            {
+                                char arpaFormat[256];
+                                MSVCRT$sprintf(arpaFormat, "%s.%s.%s.%s.in-addr.arpa", octets[3], octets[2], octets[1], octets[0]);
+                                //internal_printf("ARPA Format: %s\n", arpaFormat);
+                                query_domain(arpaFormat, DNS_TYPE_PTR, NULL, base, pSrvList);
+                            }
+                            
+                        }
+                    }
+                    else 
+                    {
+                        // resolve with NetWkstaGetInfo
+                        WKSTA_INFO_100* info = NULL;
+                        NET_API_STATUS stat = NETAPI32$NetWkstaGetInfo(clientname, 100, (LPBYTE*)&info);
+                        if (stat == NERR_Success)
+                        {
+                            internal_printf("ComputerName: %S\n", info->wki100_computername);
+                            internal_printf("ComputerDomain: %S\n", info->wki100_langroup);
+                        }
+                        else
+                        {
+                            internal_printf("ComputerName: NetWkstaGetInfo Failed; %lu\n", stat);
+                            internal_printf("ComputerDomain: NetWkstaGetInfo Failed; %lu\n", stat);
+                        }
+                        
+                    }
+
+                    internal_printf("User: %ls\n", pTmpBuf->sesi10_username);
+                    internal_printf("Active: %lu\n", pTmpBuf->sesi10_time);
+                    internal_printf("Idle: %lu\n", pTmpBuf->sesi10_idle_time);
+                    internal_printf("-------------End Session------------\n\n");
+
+                    pTmpBuf++;
+                    dwTotalCount++;
+                }
+            }
+        }
+        //
+        // Otherwise, indicate a system error.
+        //
+        else
+            BeaconPrintf(CALLBACK_ERROR, "A system error has occurred: %lu\n", nStatus);
+        //
+        // Free the allocated memory.
+        //
+        if (pBuf != NULL)
+        {
+            NETAPI32$NetApiBufferFree(pBuf);
+            pBuf = NULL;
+        }
+
+    }    
+    while (nStatus == ERROR_MORE_DATA);
+    // Check again for an allocated buffer.
+    //
+    if (pBuf != NULL)
+        NETAPI32$NetApiBufferFree(pBuf);
+    //
+    // Print the final count of sessions enumerated.
+    //
+    internal_printf("\nTotal of %lu entries enumerated\n", dwTotalCount);
+
+    // if resolveMethod is DNS, clean up
+    if (resolveMethod == 0)
+    {
+        goto END;
+    }
+
+    // DNS cleanup
+    END:
+    if(base)
+    {DNSAPI$DnsFree(base, freetype);}
+    FreeLibrary(WS);

```

Comment:
while not a crashing error this calls FreeLibrary on NULL when resolvemethod != 0
Adding a check for if(WS) will fix this

Reply:
Added a check for `if(WS)`